backend/editor/models/search_models.py

```diff
@@ -115,6 +116,130 @@ def build_cypher_query(self, param_name: str) -> CypherQuery:
         )
 
 
+class PropertyFilterSearchTerm(AbstractFilterSearchTerm):
+    filter_type: Literal["property"]
+    filter_value: str
+
+    @field_validator("filter_value")
+    @classmethod
+    def validate_filter_value(cls, filter_value: str):
+        """
+        The filter value is in the format `not:inherited:property_name:property_value`
+        where `not:` and `inherited:` and `:property_value` are optional.
+        Note that a property_name is always of format `name:lc`.
+        """
+        parsed_value = filter_value
+        if parsed_value.startswith("not:"):
+            parsed_value = parsed_value[4:]
+        if parsed_value.startswith("inherited:"):
+            parsed_value = parsed_value[10:]
+
+        assert ":" in parsed_value, "A property_name is mandatory and must contain a colon"
+
+        terms = parsed_value.split(":")
+        property_name = terms[0] + ":" + terms[1]
+
+        if not re.match(r"^[^:\\]+:[a-z]{2}$", property_name):
+            raise ValueError("Invalid property_name")
+
+        return filter_value
+
+    @computed_field
+    def negated(self) -> bool:
+        return self.filter_value.startswith("not:")
+
+    @computed_field
+    def inherited(self) -> bool:
+        filter_value = self.get_parsed_filter_value(self.negated)
+        return filter_value.startswith("inherited:")
+
+    @computed_field
+    def property_name(self) -> str:
+        filter_value = self.get_parsed_filter_value(self.negated, self.inherited)
+        terms = filter_value.split(":")
+        return terms[0] + "_" + terms[1]
+
+    @computed_field
+    def property_value(self) -> str | None:
+        filter_value = self.get_parsed_filter_value(self.negated, self.inherited)
+        terms = filter_value.split(":")
+        return ":".join(terms[2:]) if len(terms) > 2 else None
+
+    def get_parsed_filter_value(self, negated=False, inherited=False):
```

Comment:
nit: instead of this pattern I would have added a cached_property "terms" that would have returned a tuple (or dict) with not / inherited / name and value from filter_value and used it everywhere :-)

Reply:
Tracked: https://github.com/openfoodfacts/taxonomy-editor/issues/470